test/e2e/specs/block-icons.test.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Internal dependencies
+ */
+import '../support/bootstrap';
+import {
+	newPost,
+	newDesktopBrowserPage,
+	insertBlock,
+	searchForBlock,
+} from '../support/utils';
+import { activatePlugin, deactivatePlugin } from '../support/plugins';
+
+const INSERTER_BUTTON_SELECTOR = '.components-popover__content .editor-block-types-list__item';
+const INSERTER_ICON_SELECTOR = `${ INSERTER_BUTTON_SELECTOR } .editor-block-types-list__item-icon`;
+const INSPECTOR_ICON_SELECTOR = '.edit-post-sidebar .editor-block-icon__colors';
+
+async function getInnerHTML( selector ) {
+	return await page.$eval( selector, ( element ) => element.innerHTML );
+}
+
+async function getBackgroundColor( selector ) {
+	return await page.$eval( selector, ( element ) => {
+		return window.getComputedStyle( element ).backgroundColor;
+	} );
+}
+
+async function getColor( selector ) {
+	return await page.$eval( selector, ( element ) => {
+		return window.getComputedStyle( element ).color;
+	} );
+}
+
+async function getFirstInserterIcon() {
+	return await getInnerHTML( INSERTER_ICON_SELECTOR );
+}
+
+describe( 'Correctly Renders Block Icons on Inserter and Inspector', () => {
+	const dashIconRegex = /<svg.*class=".*dashicons-cart.*?">.*<\/svg>/;
```

Comment:
Across many branches, I'm getting test failures in this suite when running locally.

Example:

```
Expected value to match:
      /<svg.*class=".*dashicons-cart.*?">.*<\/svg>/
    Received:
      "<svg aria-hidden=\"true\" role=\"img\" focusable=\"false\" class=\"dashicon dashicons-editor-paragraph\" xmlns=\"http://www.w3.org/2000/svg\" width=\"20\" height=\"20\" viewBox=\"0 0 20 20\"><path d=\"M15 2H7.54c-.83 0-1.59.2-2.28.6-.7.41-1.25.96-1.65 1.65C3.2 4.94 3 5.7 3 6.52s.2 1.58.61 2.27c.4.69.95 1.24 1.65 1.64.69.41 1.45.61 2.28.61h.43V17c0 .27.1.51.29.71.2.19.44.29.71.29.28 0 .51-.1.71-.29.2-.2.3-.44.3-.71V5c0-.27.09-.51.29-.71.2-.19.44-.29.71-.29s.51.1.71.29c.19.2.29.44.29.71v12c0 .27.1.51.3.71.2.19.43.29.71.29.27 0 .51-.1.71-.29.19-.2.29-.44.29-.71V4H15c.27 0 .5-.1.7-.3.2-.19.3-.43.3-.7s-.1-.51-.3-.71C15.5 2.1 15.27 2 15 2z\"></path></svg>"
```

Also, the regular expression is prone to [greediness](https://stackoverflow.com/questions/2301285/what-do-lazy-and-greedy-mean-in-the-context-of-regular-expressions).

Reply:
I tried to debug the problem locally and if I execute the test normally they always pass. If make them slow with PUPPETEER_SLOWMO=30 they start to fail because instead of the expected block being selected the paragraph is selected. That matches with the error message you are seeing. I'm researching why the paragraph is getting selected.